photon-ml/src/main/scala/com/linkedin/photon/ml/data/GameConverters.scala

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.data
+
+import org.apache.spark.mllib.linalg.SparseVector
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.{DataFrame, Row}
+
+import com.linkedin.photon.ml.util.VectorUtils
+
+/**
+  * A collection of utility functions for converting to and from GAME datasets
+  */
+object GameConverters {
+
+  /**
+   * Standard field names
+   */
+  object FieldNames {
+    val NAME: String = "name"
+    val TERM: String = "term"
+    val VALUE: String = "value"
+    val RESPONSE: String = "response"
+    val OFFSET: String = "offset"
+    val WEIGHT: String = "weight"
+    val UID: String = "uid"
+    val META_DATA_MAP: String = "metadataMap"
+  }
+
+  // Column name for the synthesized unique id column
+  val UNIQUE_ID_COLUMN_NAME = "___photon:uniqueId___"
+
+  /**
+   * Converts a DataFrame into an [[RDD]] of type [[GameDatum]]
+   *
+   * @param data the source DataFrame
+   * @param featureShards a set of feature shard ids
+   * @param idTypeSet a set of id types expected to be found in the row
+   * @param isResponseRequired whether the response variable is expected to be found in the row. For example, if GAME
+   *   data set to be parsed is used for model training, then the response variable is expected to be found in row. If
+   *   the GAME data set is used for scoring, then we don't expect to find response.
+   * @return the [[RDD]] of type [[GameDatum]]
+   */
+  protected[ml] def getGameDataSetFromDataFrame(
+    data: DataFrame,
+    featureShards: Set[String],
+    idTypeSet: Set[String],
+    isResponseRequired: Boolean): RDD[(Long, GameDatum)] = {
+
+    // Add unique id
+    val recordsWithUniqueId = data.withColumn(UNIQUE_ID_COLUMN_NAME, monotonicallyIncreasingId)
+
+    recordsWithUniqueId.map { row: Row =>
+      val id = row.getAs[Long](UNIQUE_ID_COLUMN_NAME)
+      (id, getGameDatumFromRow(
```

Comment:
This is more of a personal style thing but I find the code more readable this way.